lib/matobj.gi

```diff
@@ -1568,3 +1568,572 @@ InstallMethod( \[\,\]\:\=, "for a matrix object, two positions, and an object",
+##  and a scalar <A>elm</A> this operation multiplies <A>elm</A> with the <A>i</A>-th column  
+##  in-place.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+InstallMethod( MultMatrixColumnLeft, "for a mutable matrix object, one column number, and an scalar",
+  [ IsMatrixObj and IsMutable, IsInt, IsObject ],
+  function( mat, column, scalar )
+    local i;
+    
+    # Checks
+    if not( 0 < column and column <= NrCols(mat) ) then
+        Print("The second argument row has to fulfill 0 < row <= NrCols(mat) ");
+        return fail;
+    fi;
+  
+    for i in [1..NrRows(mat)] do
+        mat[i,column] := scalar * mat[i,column];
+    od;
+    
+  end );
+
+############################################################################
+##
+#M  MultMatrixColumnRight( <M>, <row>, <scalar> )
+##
+##  <#GAPDoc Label="MultMatrixColumnRight">
+##  <ManSection>
+##  <Oper Name="MultMatrixColumnRight" Arg='mat,i,elm' Label=""/>
+##  
+##  <Returns>nothing</Returns>
+##
+##  <Description>
+##  <P/>
+##  For a matrix <A>mat</A> such that <A>mat</A> is mutable, a column index <A>i</A>  
+##  and a scalar <A>elm</A> this operation multiplies the <A>i</A>-th column with <A>elm</A> 
+##  in-place.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+InstallMethod( MultMatrixColumnRight, "for a mutable matrix object, one row number, and an scalar",
+  [ IsMatrixObj and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+ 
+    MultMatrixColumn(mat,row,scalar);
+    
+  end );
+  
+
+############################################################################
+##
+#M  AddMatrixRows( <M>, <row1>, <row2>, <scalar> )
+##
+##  <#GAPDoc Label="AddMatrixRows">
+##  <ManSection>
+##  <Oper Name="AddMatrixRows" Arg='mat,i,j,elm' Label=""/>
+##  
+##  <Returns>nothing</Returns>
+##
+##  <Description>
+##  <P/>
+##  For a matrix <A>mat</A> such that <A>mat</A> is mutable, row indices <A>i</A> and <A>j</A>  
+##  and a scalar <A>elm</A> this operation adds the product of <A>elm</A> and the <A>j</A>-th row
+##  to the <A>i</A>-th row in-place.
+##  Note that by default <A>elm</A> is multiplied from the left. 
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+InstallMethod( AddMatrixRows, "for a mutable matrix object, one row number, second row number, and an scalar",
+  [ IsMatrixObj and IsMutable, IsInt, IsInt, IsObject ] ,
+  function( mat, row1, row2, scalar )
+    local i;
+    
+    # Checks
+    if not( 0 < row1 and row1 <= NrRows(mat) ) then
+        Print("The second argument row has to fulfill 0 < row <= NrRows(mat) ");
+        return fail;
+    fi;
+    if not( 0 < row2 and row2 <= NrRows(mat) ) then
+        Print("The third argument row has to fulfill 0 < row <= NrRows(mat) ");
+        return fail;
+    fi;
+ 
+    if row1 <> row2 then
+        
+        for i in [1..NrCols(mat)] do
+            mat[row1,i] := mat[row1,i] + scalar * mat[row2,i];
+        od;
+        
+    else
+        MultMatrixRow(mat,row1,scalar+1);
```

Comment:
I am not sure: Can we assume that a method for `scalar + 1` is always available?